yumsync/yumrepo.py

```diff
@@ -317,16 +321,16 @@ def suppress():
 
         try:
             yb = YumBase()
+            if self.srcpkgs:
+                if not 'src' in yb.arch.archlist:
+                    yb.arch.archlist.append('src')
             repo = self._set_path(self.package_dir)
             if self.__yum_callback_obj:
                 repo.setCallback(self.__yum_callback_obj)
             yb.repos.add(repo)
             yb.repos.enableRepo(repo.id)
             with suppress():
-                # showdups allows us to get multiple versions of the same package.
-                ygh = yb.doPackageLists(showdups=True)
-            # reinstall_available = Available packages which are installed.
-            packages = ygh.available + ygh.reinstall_available
+                packages = yb.pkgSack.returnPackages()
```

Comment:
rather than  comparing repo-packages with installed/available/.... packages on the yumsync server, i thought it would be enough to simply create packagelist from all packages in repo and use same proc for that like  upstream yumBase.doPackageLists does (compare http://yum.baseurl.org/api/yum-3.2.27/yum-pysrc.html#YumBase.doPackageLists, l1888)